### ISSUE TITLE
About domains in code 

### ISSUE THREAD

**Issue by HOLYlmx** [vihari/CSD#2]
'''domains (tensor): tf tensor with domain index of dim 1 -- set to all zeros when testing
  domains = torch.nn.functional.one_hot(domains, num_domains) '''

how should i set this 'domains' parameter, i have num_domains = 3 in my code ,if i want to deploy csd in my code, what size should this domains get ? thanks a lot.

**Comment by vihari** [comment]
Then make a placeholder or a vector which contains domain index (ranging from 0 to 2 here) for each example in the batch.
For example, if the batch contains 3 examples from domains 2, 0, 1 respectively then the domains vector is a pytorch tensor of size 3: [2, 0, 1].

**Comment by HOLYlmx** [comment]
Thank you so much, if my input is " input_data = torch.cat([src1_img_real, src1_img_fake, src2_img_real, src2_img_fake, src3_img_real, src3_img_fake]", and batch_size = 10, so the corresboding domains should be one dim tensor like:[0,0,1,1,2,2,0,0,1,1,2,2.......]with size 60. can I set domains as a fixed parameter like that?   ＾＿＾

**Comment by vihari** [comment]
Yes, that looks correct.
If the order of the concat remains the same in all the batches, you can leave domains to be a constant vector.

**Comment by HOLYlmx** [comment]
Thanks　∩＿∩, i have another question about the batch_size in CSD , my input contains 60 imgs , so the batch should be embds[0] (60), or should i change it to embds[0]/6   (10)    , thanks for your patience again，its very helpful for me .
# B x K
batch_size = embeds.shape[0]

**Comment by vihari** [comment]
Why would it be 10? Should be 60, the length of your input.

**Comment by HOLYlmx** [comment]
hh , i see , thank you :+1:
